### ISSUE TITLE
Error: Compilation produced no files satisfying /homedir/project/target/release/deps/project-*.s, this is a bug

### ISSUE THREAD

**Issue by luctius** [pacak/cargo-show-asm#9]
Trying `cargo asm --bin <project> <function_name>` on my project fails with the above error. The listed path is wrong since I'm cross-compiling for thumbv7em-none-eabihf.

The correct path should be /homedir/project/target/thumbv7em-none-eabihf/release/deps/project-*.s.

**Comment by pacak** [comment]
It would be great if you could help me with reproducing your problem, creating a new project from scratch and cross compiling from x86_64 to aarch64 seems to work.

```
% cargo new --bin hello
% cd hello
% cargo asm --bin hello hello::main --target aarch64-unknown-linux-gnu -v
   Compiling hello v0.1.0 (/home/pacak/rust/hello)
     Running `/home/pacak/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc --crate-name hello --edition=2021 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 --emit asm -C llvm-args=-x86-asm-syntax=intel -C debuginfo=2 --target aarch64-unknown-linux-gnu -C linker=aarch64-linux-gnu-gcc -C metadata=7be3ebfda4b85da4 -C extra-filename=-7be3ebfda4b85da4 --out-dir /home/pacak/rust/hello/target/release/deps -L dependency=/home/pacak/rust/hello/target/release/deps`
    Finished release [optimized] target(s) in 0.19s
hello::main:
.Lfunc_begin2:
        .cfi_startproc
        sub sp, sp, #64
        stp xzr, x30, [sp, #40]
        .cfi_def_cfa_offset 64
        .cfi_offset w30, -16
.Ltmp5:
        adrp x8, .L__unnamed_2
        mov w9, #1
        add x8, x8, :lo12:.L__unnamed_2
.Ltmp6:
        mov x0, sp
.Ltmp7:
        stp xzr, xzr, [sp, #16]
        stp x8, x9, [sp]
        adrp x8, .L__unnamed_3
.Ltmp8:
        add x8, x8, :lo12:.L__unnamed_3
.Ltmp9:
        str x8, [sp, #32]
.Ltmp10:
        bl std::io::stdio::_print
.Ltmp11:
        ldr x30, [sp, #48]
        add sp, sp, #64
        ret
.Ltmp12:
.Lfunc_end2:
        .size   hello::main, .Lfunc_end2-hello::main
        .cfi_endproc
```

**Comment by luctius** [comment]
Sure, apologies that it is slightly convoluted.

Create project and enter
```
cargo new --bin hello
cd hello
```
Modify main.rs
```
echo '#![no_main]                                                                               
#![no_std]

use panic_halt as _;

use cortex_m_rt::entry;

#[entry]
fn main() -> ! {
    loop {}
}' > src/main.rs
```
Set-up Cargo.toml
```
echo 'cortex-m = "0.6.2"                                                                        
cortex-m-rt = "0.6.11"
panic-halt = "0.2.0"' >> Cargo.toml
```

get necessary memory.x and set-up .cargo/config
```
wget https://raw.githubusercontent.com/stm32-rs/stm32g4xx-hal/master/memory.x
mkdir .cargo && cd .cargo
wget https://raw.githubusercontent.com/stm32-rs/stm32g4xx-hal/master/.cargo/config 
```
Finaly run cargo asm
`cargo asm`

**Comment by luctius** [comment]
Yes, the cross branch works!

As per your request, a slightly less useless example:

Create project and enter
```
cargo new --bin hello
cd hello
```
Modify main.rs
```
echo '#![no_main]
#![no_std]

use defmt_rtt as _;
use panic_halt as _;

use cortex_m_rt::entry;

#[inline(never)]
fn print() {
    defmt::error!("idle loop");
}

#[entry]
fn main() -> ! {
    loop {
        print();
    }
}' > src/main.rs
```
Set-up Cargo.toml
```
echo 'cortex-m = "0.6.2"                                                                        
cortex-m-rt = "0.6.11"
panic-halt = "0.2.0"
defmt = "0.3.0"
defmt-rtt = "0.3.0"' >> Cargo.toml
```

get necessary memory.x and set-up .cargo/config
```
wget https://raw.githubusercontent.com/stm32-rs/stm32g4xx-hal/master/memory.x
mkdir .cargo
echo '[target.thumbv7em-none-eabihf]
runner = "arm-none-eabi-gdb -q -x openocd.gdb"
rustflags = [
  "-C", "linker=flip-link",
  "-C", "link-arg=-Map=.linker.map",
  "-C", "link-arg=-Tlink.x",
  "-C", "link-arg=-Tdefmt.x",
]

[build]
target = "thumbv7em-none-eabihf"' > .cargo/config
```

`cargo asm hello::print`

**Comment by pacak** [comment]
Should be fixed here 10fc64fa77c20235881141aa7b79f7be304fdf70, feel free to reopen if something is still broken
